src/lib.rs

```diff
@@ -2002,6 +2002,9 @@ pub enum SymbolKind {
     Event = 24,
     Operator = 25,
     TypeParameter = 26,
+
+    // Capturing all unknown enums by this lib.
+    Unknown = 255,
```

Comment:
So I argued for adding an unknown variant so that it would be possible to conform to LSP if more variants were added in the future (otherwise servers and clients may break until they get an updated version of this crate). And reading the docs of `symbol.symbolKind` in https://github.com/Microsoft/language-server-protocol/blob/gh-pages/specification.md#workspaceclientcapabilities-define-capabilities-the-editor--tool-provides-on-the-workspace we can see that there should be a way to handle that case gracefully.

If you agree with me with that I think we might as well capture the value of the unknown variant as `Unknown(u8/u32)`. Alternatively we could add a wrapping enum around `SymbolKind` as

```rust
enum MaybeSymbolKind {
    SymbolKind(SymbolKind),
    Unknown(u32),
}
```

that way we still have an exhaustive `SymbolKind` for all (known) valid variants. Might not be worth it though.

All that said, I don't know if LanguageClient-Neovim sets `symbol.symbolKind`? If it does not then cquery is definitely not compliant here (since ONLY the first defined variants should be sent in that case). If LanguageClient-Neovim does set `symbol.symbolKind` set then I think it should handle the custom variant gracefully, HOWEVER cquery is still non-compliant since LSP could be extended to use that value as a variant in the future

Reply:
LanguageClient-neovim deosn't send `symbol.symbolKind`.

Will make an update to include the unknown value in enum.

Reply:
Hmm, realized the enum is using enum_primitive and doesn't support embed the value in variant.

Reply:
Right :/ Well, it is fine to just ignore the value then. I guess https://github.com/serde-rs/serde/issues/912 is needed to do this in a convenient way.